cns/fsnotify/fsnotify.go

```diff
@@ -206,3 +224,34 @@ func (w *watcher) releaseIP(ctx context.Context, podInterfaceID, containerID str
 	}
 	return errors.Wrap(w.cli.ReleaseIPs(ctx, *ipconfigreq), "failed to release IP from CNS")
 }
+
+// call GetEndpoint API to get the state and then remove assiciated HNS
+func (w *watcher) deleteEndpoint(ctx context.Context, containerid string) error {
+	endpointResponse, err := w.cli.GetEndpoint(ctx, containerid)
+	if err != nil {
+		return errors.Wrap(err, "failed to read the endpoint from CNS state")
+	}
+	for _, ipInfo := range endpointResponse.EndpointInfo.IfnameToIPMap {
+		hnsEndpointID := ipInfo.HnsEndpointID
+		// we need to get the HNSENdpoint via the IP address if the HNSEndpointID is not present in the statefile
+		if ipInfo.HnsEndpointID == "" {
+			if hnsEndpointID, err = hnsclient.GetHNSEndpointbyIP(ipInfo.IPv4, ipInfo.IPv6, DefaultNetworkID); err != nil {
+				return errors.Wrap(err, "failed to find HNS endpoint with id")
+			}
+		}
+		w.log.Info("deleting HNS Endpoint with id ", zap.String("id", hnsEndpointID))
+		if err := hnsclient.DeleteHNSEndpointbyID(hnsEndpointID); err != nil {
+			return errors.Wrap(err, "failed to delete HNS endpoint with id "+ipInfo.HnsEndpointID)
+		}
+	}
```

Comment:
this should be in windows specific file. may be in another PR.. lets remove `hnsclient_linux.go` and `hnsclient_windows.go` and instead have `endpoint_linux.go` and `endpoint_windows.go` and abstract hns changes in `endpoint_windows.go`

Reply:
Will make a TODO comment and do another PR to address this since it affects the multitenancy too and it's beyond this PR goal